src/lib.rs

```diff
@@ -77,7 +77,7 @@ pub trait Hash: Copy + Clone + PartialEq + Eq +
     ops::Index<ops::RangeTo<usize>, Output = [u8]> +
```

Comment:
Aren't the `Index` impls kinda redundant if you have `Borrow<[u8]>`?

Reply:
Don't think so... We can have both, just like `Vec` does.

Reply:
Ok, makes sense if you think of it as a slim wrapper around a byte array.